src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -289,17 +293,38 @@ private Request BuildRequest(HttpRequest httpRequest, RequestMethodBuildContext
 
         private Parameter[] BuildMethodParameters(IReadOnlyDictionary<RequestParameter, Parameter> allParameters)
         {
-            List<Parameter> methodParameters = new();
+            List<Parameter> requiredParameters = new();
+            List<Parameter> optionalParameters = new();
+            List<Parameter> bodyParameters = new();
             foreach (var (requestParameter, parameter) in allParameters)
             {
                 // Grouped and flattened parameters shouldn't be added to methods
                 if (IsMethodParameter(requestParameter))
                 {
-                    methodParameters.Add(parameter);
+                    // sort the parameters by the following sequence:
```

Comment:
This change will break parameter ordering in all existing HLC clients.

Reply:
Thanks. I thought the ordering is for HLC as well. I'll split the logic.

Reply:
according to https://github.com/Azure/autorest.csharp/issues/1771 this should apply to all .net sdk